wallet/wallet.go

```diff
@@ -332,84 +352,93 @@ func (w *SingleAddressWallet) Redistribute(cs consensus.State, outputs int, amou
 		}
 	}
 	if outputs <= 0 {
-		return types.Transaction{}, nil, nil
-	}
-
-	// prepare all outputs
-	var txn types.Transaction
-	for i := 0; i < int(outputs); i++ {
-		txn.SiacoinOutputs = append(txn.SiacoinOutputs, types.SiacoinOutput{
-			Value:   amount,
-			Address: w.Address(),
-		})
+		return nil, nil, nil
 	}
 
 	// desc sort
 	sort.Slice(utxos, func(i, j int) bool {
 		return utxos[i].Value.Cmp(utxos[j].Value) > 0
 	})
 
-	// estimate the fees
-	outputFees := feePerByte.Mul64(uint64(len(encoding.Marshal(txn.SiacoinOutputs))))
-	feePerInput := feePerByte.Mul64(BytesPerInput)
+	// prepare all outputs
+	var txns []types.Transaction
+	var toSign []types.Hash256
+
+	for outputs > 0 {
+		var txn types.Transaction
+		for i := 0; i < outputs && i < redistributeBatchSize; i++ {
+			txn.SiacoinOutputs = append(txn.SiacoinOutputs, types.SiacoinOutput{
+				Value:   amount,
+				Address: w.Address(),
+			})
+		}
 
-	// collect outputs that cover the total amount
-	var inputs []SiacoinElement
-	want := amount.Mul64(uint64(outputs))
-	var amtInUse, amtSameValue, amtNotMatured types.Currency
-	for _, sce := range utxos {
-		inUse := w.isOutputUsed(sce.ID) || inPool[sce.ID]
-		matured := cs.Index.Height >= sce.MaturityHeight
-		sameValue := sce.Value.Equals(amount)
-		if inUse {
-			amtInUse = amtInUse.Add(sce.Value)
-			continue
-		} else if sameValue {
-			amtSameValue = amtSameValue.Add(sce.Value)
-			continue
-		} else if !matured {
-			amtNotMatured = amtNotMatured.Add(sce.Value)
-			continue
+		// estimate the fees
+		outputFees := feePerByte.Mul64(uint64(len(encoding.Marshal(txn.SiacoinOutputs))))
+		feePerInput := feePerByte.Mul64(BytesPerInput)
+
+		// collect outputs that cover the total amount
```

Comment:
if `fundTransaction` is no longer selecting random outputs, couldn't we re-use `FundTransaction` here?

Reply:
Unfortunately not. That `NOTE` wasn't quite right. Due to the way we pick the inputs we can't reuse `FundTransaction`.